django/thunderstore/api/cyberstorm/views/package_versions.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import serializers
+from rest_framework.generics import ListAPIView, get_object_or_404
+
+from thunderstore.api.utils import CyberstormAutoSchemaMixin
+from thunderstore.repository.models import Package
+
+
+class CyberstormPackageVersionSerializer(serializers.Serializer):
+    version_number = serializers.CharField()
+    datetime_created = serializers.DateTimeField(source="date_created")
+    download_count = serializers.IntegerField(min_value=0, source="downloads")
+    download_url = serializers.CharField(source="full_download_url")
+    install_url = serializers.CharField()
+
+
+class PackageVersionsAPIView(CyberstormAutoSchemaMixin, ListAPIView):
+    """
+    Return a list of available versions of the package.
+    """
+
+    serializer_class = CyberstormPackageVersionSerializer
+
+    def get_queryset(self):
+        package = get_object_or_404(
+            Package.objects.active(),
+            namespace__name=self.kwargs["namespace_id"],
+            name__iexact=self.kwargs["package_name"],
```

Comment:
I keep seeing `iexact` used for querying the name field on packages/package versions, and while that's valid in terms of the business logic, it shouldn't be necessary to build into the queries.

The gist of the issue is that the `name` field on the table currently isn't case-invariant, but it _should_ be. That said, in the present moment it means this query is potentially wrong, but on the other hand if we migrate the field to the case-invariant version this becomes unnecessary.

So it would seem like in both scenarios, iexact querying is actively harmful at worst or pointless at best?

Not a blocker, just something I noticed a few times now & finally wrote down. Applies to many other endpoints outside of this PR too.

Reply:
The rationale was that the client side currently casts the names to lowercase in the hope of making the requests more cacheable, thus the queries are made as case insensitive. This is why it's done in multiple endpoints. Does that idea not apply for this endpoint, or in general?

Reply:
Ah right, yeah makes sense then. Just means we'll have a pointless `iexact` use when the field is updated, but that's not a big deal at the end of the day.